### PR TITLE
Modeler: New UI Documentation

### PR DESCRIPTION
**Pull request by DiogoTx**

### CODE REVIEW

File: docs/_data/sidebars/omnia3_sidebar.yml
Comment (davidcaldas) [inline docs/_data/sidebars/omnia3_sidebar.yml:1]: Change PRIMAVERA (v9) to PRIMAVERA (v10)

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+title: Themes
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Change to Pages

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
+---
+title: Themes
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Change to Components

File: docs/pages/omnia3/Modeler/omnia3_modeler_cssStyles.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: Themes
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_cssStyles.md]: Change to CSS Styles

File: docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+This feature allows you to add Javascript code that can be used on other UI code expressions.
+
+On each model you can have multiple Scripts. Each script is composed by a class, named after the script. 
+
+
+```Javascript
+  /** 
+ * You can invoke these class methods in the expression and behaviours code. Use the 'helpers' object to do it.
+ */
+class HelpersScripts {
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md]: Change "Helpers" to other name (ex: MyLib)

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+On OMNIA Platform models, you can add your own Components to be used on [Pages](omnia3_modeler_pages.md).
+
+These components have the same purpose as the ones available on Forms and Dashboards but, unlike those, they can be fully controlled by the modeler.
+
+Components can be visual (e.g. inputs) or functional. Users can interact with the visual components directly, while functional components are used to execute multiple processes behind the screen (e.g. fetch data, act as container to render other elements).
+
+OMNIA Platform contains a small number of base components, and new ones can be added to each model, according to its needs. Multiple packages with different components can be uploaded, and you can develop your own components. To start, we suggest using the package made available by the OMNIA Platform, that you can find [here](https://github.com/OMNIALowCode/omnia-base-components).
+
+## 2. Base Components
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Base components can be confused by omnia base components. Perhaps "internal"

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+This Component only has a configurable attribute, named _selector_. This attribute is used to define the Enumeration Selector whose configuration will be retrieved.
+
+Component Behaviours:
+
+This Component only has a Behaviour type, named _onLoad_. This behaviour is executed after Enumerator data was retrieved, and can be used to store that data to be rendered on a select Component.
+
+## 3. Modeling Components
+
+### How to add a new Components package?
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md:229]: Show omnia base component full add example inc. link

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+</p>
+
+You can drag them to the Page preview area and change their Attributes, Behaviours and Styles. Each Component exposes a different set of Attributes, Behaviours and Styles that you can configure while modeling.
+
+Focusing on Attributes and Styles, its value can be set in different ways:
+
+| Value Source | Description                                                      |
+| ------------ | ---------------------------------------------------------------- |
+| Direct       | Value is set directly, with a static value                       |
+| Binding      | The dataSource instance where the entity is being created/edited |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Wrong explanation

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+| Expression   | Value is computed using a Javascript code expresion              |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Add a note: expressions are computed on every render of react. Must be simple operations.

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+Fill the following parameters:
+
+- **Name**: the name of the Variable (needs to be unique within the Page);
+- **Description**: the textual explanation of the Variable purpose (can be used as development documentation).
+- **Kind**: the kind of the variable. Can be of type Primitive, an OMNIA definition (Agent, Resource, ...) or a Component Type. These types are exposed by the component packages;
+- **Type**: the type of the variable
+- **Is required value?**: define if the variable must have a value;
+- **Is a list of records?**: define if the variable contains more than one record of the given type;
+- **The initial value of the variable**: the value the variable is initialized with.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Only Primitives

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+#### How to use a variable?
+
+Variables can be used on Code Expressions within the page. They are available on _variables_ object.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Add binding exemple

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md:1]: Missing EditorSidebar explanation
Missing talk about default components default properties:
isHidden
groupElements
classesStyles
cssStyle

File: docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+sidebar: omnia3_sidebar
+permalink: omnia3_modeler_scripts_UI.html
+folder: omnia3
+---
+
+## 1. Introduction
+
+This feature allows you to add Javascript code that can be used on other UI code expressions.
+
+On each model you can have multiple Scripts. Each script is composed by a class, named after the script.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md]: Review

File: docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+### How to add a new Script?
+
+To add a new script go to the Modeling area, find the **_User Interface / Scripts_** option on the menu and access it. This will take you to your scripts management dashboard.
+
+Now select _Add new_ and fill in the following information:
+
+- **Name**: the name of the script (needs to be unique within the model);
+- **Description**: the textual explanation of the script (can be used as development documentation);
+- **Order**: the order that the script class will be loaded;
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md]: This is not necessary

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+Components can be visual (e.g. inputs) or functional. Users can interact with the visual components directly, while functional components are used to execute multiple processes behind the screen (e.g. fetch data, act as container to render other elements).
+
+OMNIA Platform contains a small number of internal components, and new ones can be added to each model, according to its needs. Multiple packages with different components can be uploaded, and you can develop your own components. To start, we suggest using the package made available by the OMNIA Platform, that you can find [here](https://github.com/OMNIALowCode/omnia-base-components).
+
+## 2. Components Structure
+
+On the modeler perspective, a Component is composed by the following definitions:
+
+### Properties
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Change to Attributes

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+## 1. Introduction
+
+On OMNIA Platform models, you can add your own Components to be used on [Pages](omnia3_modeler_pages.md).
+
+These components have the same purpose as the ones available on Forms and Dashboards but, unlike those, they can be fully controlled by the modeler.
+
+Components can be visual (e.g. inputs) or functional. Users can interact with the visual components directly, while functional components are used to execute multiple processes behind the screen (e.g. fetch data, act as container to render other elements).
+
+OMNIA Platform contains a small number of internal components, and new ones can be added to each model, according to its needs. Multiple packages with different components can be uploaded, and you can develop your own components. To start, we suggest using the package made available by the OMNIA Platform, that you can find [here](https://github.com/OMNIALowCode/omnia-base-components).
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]:
```suggestion
OMNIA Platform contains a small number of internal components, and new ones can be added to each model, according to its needs. Multiple packages with different components can be uploaded, and you can develop your own components. To start, we suggest using the package made available by the OMNIA Platform, that you can find <a href="https://github.com/OMNIALowCode/omnia-base-components" target="_blank">here</a>.
```

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+
+An attribute can be required, and have a default value to be suggested when modeling. Depending on the attribute _direction_ configuration, its value can be only read (Outbound direction) or changed (Inbound direction), or both (TwoWay direction).
+
+Some scenarios where attributes can be applied are setting/getting the component value or if it rendered as readOnly. An attribute value can be set in multiple ways.
+
+### Events
+
+Component Events are code expressions executed on different moments when interacting with the component.
+
+When the event is executed, it receives a list of parameters on object _params_. These parameters can be of multiple types, and contains the values of the _Outbound_ and _TwoWay_ properties.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: "and contains the values of the _Outbound_ and _TwoWay_ properties" ?

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+
+## 2. Components Structure
+
+On the modeler perspective, a Component is composed by the following definitions:
+
+### Properties
+
+The component attributes contain data relevant to define the component and change how it renders and behaves on a page. A component can have multiple attributes, and they can be different between components.
+
+An attribute can be required, and have a default value to be suggested when modeling. Depending on the attribute _direction_ configuration, its value can be only read (Outbound direction) or changed (Inbound direction), or both (TwoWay direction).
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: The attribute can be defined as Outbound, Inbound, and TwoWay. Outbound attributes are read-only and only the component can change the value. Inbound attributes can only be changed by Page "interaction". TwoWay attributes can be changed for all.

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+
+Some scenarios where attributes can be applied are setting/getting the component value or if it rendered as readOnly. An attribute value can be set in multiple ways.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: rephrase

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+
+### Events
+
+Component Events are code expressions executed on different moments when interacting with the component.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Events are user actions such as key press, clicks, mouse movements, etc,...
The modeler can add custom javascript code the will be executed when event raises. 
When the event is executed, it receives a list of parameters on object _params_. Each component defines the parameters for each event and can have multiple types.

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+
+The Methods have entry parameters, whose values should be sent when calling them. A component can have as many Methods as necessary.
+
+## 2. Internal Components
+
+OMNIA Platform has four internal components:
+
+### forEach
+
+The forEach Component iterates a collection, and renders other Components and data for each record of the collection.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: The forEach is a functional Component that iterates a collection and renders children Components for each record of the collection

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+| Method    | Description                                          |
+| --------- | ---------------------------------------------------- |
+| refresh   | Method that executes the query to retrieve new data  |
+| exportCSV | Method that requests to the API a data export to CSV |
+
+### entityForm
+
+The entityForm Component renders a Form composed by other Components inside a page.
+
+A common scenario for this component is to show a Form to create/edit records.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: The entityForm is a functional Component use to manipulate modeling entities data (create/update/read/delete).

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+| OnSaveError       | Executed when the Save operation is finished with error. Can be used to show the error to the user                        |
+| OnDelete          | Executed when the Delete operation is finished with success. Can be used to show a success message and redirect the user  |
+| OnDeleteError     | Executed when the Delete operation is finished with error. Can be used to show the error to the user                      |
+| OnDestroy         | Executed when the Destroy operation is finished with success. Can be used to show a success message and redirect the user |
+| OnDestroyError    | Executed when the Destroy operation is finished with error. Can be used to show the error to the user                     |
+| OnLoad            | Executed when the data Load operation is finished with success. Can be used to remove a loader                            |
+| OnLoadError       | Executed when the data Load operation is finished with error. Can be used to show the error to the user                   |
+| OnUpdateData      | Executed when the Update operation is finished with success                                                               |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md:141]: "Can be used to show the error to the user"

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+| OnUpdateDataError | Executed when the Update operation is finished with error. Can be used to show the error to the user                      |
+
+- Methods:
+
+| Method | Description                                                                                                                                       |
+| ------ | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| save   | Method that makes a request to the API to save the form data. It has 2 optional parameters related to the state machine: _decision_ and _comment_ |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Missing  delete and destroySensitiveData

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 23]
```diff
@@ -0,0 +1,189 @@
...
+| definition       | Inbound   | Reference | The modeled entity the Form is referencing                                                |
+| dataSource       | Inbound   | Text      | The dataSource instance where the entity is being created/edited                          |
+| code             | Inbound   | Text      | The instance of the definition being edited                                               |
+| data             | Outbound  | Object    | The instance data contained in the form component                                         |
+| hasChanges       | Outbound  | Boolean   | Flag to indicate if the data contained in the form has changed                            |
+| useTemporary     | Inbound   | Boolean   | Flag to indicate if the create/edit operation is using a Temporary                        |
+| hasErrors        | Outbound  | Boolean   | Flag to indicate if the form has errors                                                   |
+| state            | Outbound  | Object    | The state machine state of the entity                                                     |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Type types of all known objects

File: docs/pages/omnia3/Modeler/omnia3_modeler_queries.md [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+        vw_company.identifier,
+        vw_company._created_at,
+        vw_company._updated_at,
+        vw_company.sales
+    FROM vw_company
+```
+
+### _Output Structure_
+
+On Advanced Queries, it's possible to define the output structure. This structure can be used later when creating a new list, to define the columns.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_queries.md]: Refer label, kind and type
runQuery get this information to put in columns attribute

File: docs/pages/omnia3/Modeler/omnia3_modeler_queries.md [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+
+To set the query output structure, on the advanced query editor navigate to Columns tab and click on the _Auto-fill_ button:
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_queries.md]: Auto-fill only in system

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+| selector | Inbound   | Reference | This attribute is used to define the Enumeration Selector whose configuration will be retrieved |
+| entries  | Outbound  | Object    | A list containing the entries of the Enumeration                                                |
+
+- Events:
+
+| Event  | Description                                                                                                                                          |
+| ------ | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+| OnLoad | Code executed when Enumerator data is retrieved from API, before being rendered. Can be used to store that data to be rendered on a select Component |
+
+## 3. Modeling Components
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md:169]: Change Modeling to Manage

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+
+### How to edit a Components package?
+
+By accessing **_User Interface / Components_** in the sidebar and selecting one from the list.
+
+You can upload a new version of the components package.
+
+## 4. Components Usage
+
+Components are available to be used on modeled Pages. See how its done [here](omnia3_modeler_pages.md).
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Add 5. Create new Components
redirect to git

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 28]
```diff
@@ -0,0 +1,67 @@
...
+sidebar: omnia3_sidebar
+permalink: omnia3_modeler_page_generators.html
+folder: omnia3
+---
+
+## 1. Introduction
+
+You have full freedom to design your model [Pages](omnia3_modeler_pages.md) as you want.
+
+But sometimes we just need to quicky create a Page, to be used as a starting point or on a Demo.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: quickly

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 29]
```diff
@@ -0,0 +1,67 @@
...
+
+When the template is executed, it generates a page that is ready to be used. The result includes a number of Components, Variables and Url Parameters. To generate a page the [Omnia Base Components](https://github.com/OMNIALowCode/omnia-base-components) package must be imported.
+
+When a page is generated using a template, the following variables are created:
+
+- Create or Update an Entity:
+
+| Variable                 | Type    | Description                                                                                  |
+| ------------------------ | ------- | -------------------------------------------------------------------------------------------- |
+| RedirectTo               | Text    | The OMNIA page address to redirect the user                                                  |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: RedirectTo change to redirectTo
When entity is save or deleted

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+| ShowErrors               | Boolean | Boolean to indicate if page errors should be visible                                         |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: ShowErrors change to showErrors

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 31]
```diff
@@ -0,0 +1,67 @@
...
+| stateMachineDropDownOpen | Boolean | Boolean to control if the state machine dropdown is opened                                   |
+| optionsDropDownOpen      | Boolean | Boolean to control if the options button dropdown is opened                                  |
+| canDelete                | Boolean | Boolean to control if the user has privileges to request a entity deletion                   |
+| canDestroy               | Boolean | Boolean to control if the user has privileges to request a entity sensitive data destruction |
+
+- List entities returned by a Query:
+
+| Variable         | Type    | Description                                                                          |
+| ---------------- | ------- | ------------------------------------------------------------------------------------ |
+| CurrentPage      | Integer | The number of the page being currently shown in the list                             |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: CurrentPage change to currentPage

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 32]
```diff
@@ -0,0 +1,67 @@
...
+| PageSize         | Integer | The number of records being requested when executing the query                       |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: PageSize change to pageSize

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 33]
```diff
@@ -0,0 +1,67 @@
+---
+title: Pages
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: Chante to Page Generators

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 34]
```diff
@@ -0,0 +1,255 @@
...
+    https://www.example.com/widgets?color=blue&size=large
+```
+
+We have two Url parameters to define in our page: color and size.
+
+Unlike Variables, a Url parameter value is automatically set from the Url used to access the page. The value can be read but not changed.
+
+### Refs
+
+Refs are responsible by exposing _Outbound_ and _TwoWay_ Component attributes so their value can be used on a Page.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Only Outbound

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md [hunk 35]
```diff
@@ -0,0 +1,67 @@
...
+
+When a page is generated using a template, the following variables are created:
+
+- Create or Update an Entity:
+
+| Variable                 | Type    | Description                                                                                  |
+| ------------------------ | ------- | -------------------------------------------------------------------------------------------- |
+| RedirectTo               | Text    | The OMNIA page address to redirect the user                                                  |
+| ShowErrors               | Boolean | Boolean to indicate if page errors should be visible                                         |
+| IsLoading                | Boolean | Boolean to indicate if the page is loading. A loader is shown if true                        |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages_generator.md]: IsLoading change to isLoading

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 36]
```diff
@@ -0,0 +1,255 @@
...
+
+They are available to be used on code expressions and on page attributes.
+
+### Behaviours
+
+OMNIA Platform Pages only have one behaviour available: Initialize.
+
+As on other modeling features like Forms or Dashboards, this Behaviour will be executed when accessing the page.
+
+Other Behaviours that are usually available (e.g. Formula, On Change, Before Save) should now be implemented on [Components](omnia3_modeler_components.md).
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Behaviour or Event?

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 37]
```diff
@@ -0,0 +1,255 @@
...
+_descriptionInput_OnChange({ context, variables, urlParameters, refs, indexes, currentElement, currentIndex }, params){
+    //add your code here
+ }
+```
+
+Component Behaviours contains the properties available on Page Behaviours and adds the following:
+
+| Property       | Description                                                                                                |
+| -------------- | ---------------------------------------------------------------------------------------------------------- |
+| indexes        | When the behaviour is in a nested collection, this property contains all indexes of the current element path |
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Improve with more detail (forEach) and example

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 38]
```diff
@@ -0,0 +1,255 @@
...
+| currentElement | When the behaviour is in a collection, this property contains the current entry data                       |
+| currentIndex   | When the behaviour is in a collection, this property contains the index of the collection entry            |
+| params         | The values of the _Outbound_ and _TwoWay_ properties of the Component                                      |
+
+### Scripts
+
+Each page has a _Scripts_ code area that can be used to define JavaScript functions to be used on multiple code expressions within the page.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md:132]: Can call modeling scripts

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 39]
```diff
@@ -0,0 +1,255 @@
...
+
+The script is composed by a class named after the page. The custom code must be added inside this class.
+
+```Javascript
+class CompanyFormPageScripts {
+
+//add your code here
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: missing showAlert

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 40]
```diff
@@ -0,0 +1,255 @@
...
+
+Fill the following parameters:
+
+- **Name**: the name of the Variable (needs to be unique within the Page);
+- **Description**: the textual explanation of the Variable purpose (can be used as development documentation).
+- **Kind**: the kind of the variable. Can be of type Primitive, an OMNIA definition (Agent, Resource, ...) or a Component Type. These types are exposed by the component packages;
+- **Type**: the type of the variable
+- **Is required value?**: define if the variable must have a value;
+- **Is a list of records?**: define if the variable contains more than one record of the given type;
+- **The initial value of the variable**: the value the variable is initialized with.
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Only Primitive

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 41]
```diff
@@ -0,0 +1,255 @@
...
+On modeler, navigate to a page and on the right side menu open option **_Properties_**, and _Styles_ area will be available.
+
+To use a global style, simply introduce the CSS classes on property _classesStyles_.
+
+### How to add a page style?
+
+On modeler, navigate to a page and on the right side menu open option **_Properties_**, and _Styles_ area will be available.
+
+You can add page specific styles by introducing them on property _styles_.These styles can then be used in the page or page components.
+
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md:302]: How to use a global script?
How to add a page script?

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md [hunk 42]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+| Property      | Description                                                                            |
+| ------------- | -------------------------------------------------------------------------------------- |
+| context       | This property contains information about the current user and the operation being made |
+| variables     | The values of the variables modeled on the page                                        |
+| urlParameters | The values of the Url parameters set when acessing the page                            |
+| refs          | The _Outbound_ attribute values for each Component located on the page                 |
+
+- Component Behaviour
```
Comment (davidcaldas) [inline docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Events

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md (hunk 17 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Give examples in a bullet list. Describe type bindings, expressions, etc

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md (hunk 15 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Do not describe that vales are sent on params. That is a concern of the page, not the component

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md (hunk 23 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Describe known types on a separate chapter

File: docs/pages/omnia3/Modeler/omnia3_modeler_components.md (hunk 27 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_components.md]: Suggest clone our base components repo

File: docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md (hunk 12 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md]: Remove "Order" bullet

File: docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md (hunk 11 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_scripts_UI.md]: Enforce code reusability

File: docs/pages/omnia3/Modeler/omnia3_modeler_queries.md (hunk 25 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_queries.md]: Describe that even though auto-fill is only available in system, queries should always be typed

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md (hunk 36 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_pages.md]: Behaviours are implemented on the page, for each component event.

Introduce "Element" figure. Element references a component

File: docs/pages/omnia3/Modeler/omnia3_modeler_pages.md (hunk 38 above)
Comment (DiogoTx) [reply docs/pages/omnia3/Modeler/omnia3_modeler_pages.md:132]: Call external scripts

**Pull request merged**